### DISCUSSION TITLE
Invalid type conversion on watchContractEvent after upgrade

### DISCUSSION THREAD

**Discussion by wslyvh** [wevm/wagmi#3057] [Q&A]
Hi,

I upgraded Wagmi, CLI and Viem to their latest version on my existing boilerplate project.
But when I try to `wagmi generate` it generates invalid type conversions for watchContractEvent.

Disabling watchContractEvent in wagmiConfig generated a valid ABI file.

Here's a short snippet
Conversion of type '{ address?: `0x${string}` | `0x${string}`[] | undefined; chainId?: number | undefined; eventName?: InferEventName<TAbi, TEventName> | undefined; abi: readonly [{ ...; }, ... 20 more ..., { ...; }]; }' to type 'WatchContractEventConfig<TAbi, TEventName>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ address?: `0x${string}` | `0x${string}`[] | undefined; chainId?: number | undefined; eventName?: InferEventName<TAbi, TEventName> | undefined; abi: readonly [{ ...; }, ... 20 more ..., { ...; }]; }' is not comparable to type 'Pick<WatchContractEventParameters<TAbi, TEventName>, "address" | "abi" | "eventName">'.
    Types of property 'abi' are incompatible.

And the upgrade commit
https://github.com/wslyvh/nexth/commit/a8cf7d2d9573c27a2f781784f694abb6a7ee04a9

Versions
viem: 1.12.2
wagmi: 1.4.2
cli: 1.5.2
ts 5.2.2 

Any thoughts?

**Comment by jrfrantz** [comment]
very curious if you were able to resolve this, @wslyvh !

**Reply by wslyvh** [reply]
I didn't no. Sorry.

I just disabled the watchContractEvent action from my wagmi config as I didn't really need it.
